packages/legend-query-builder/src/stores/QueryBuilderValueSpecificationBuilder.ts

```diff
@@ -158,6 +182,39 @@ export const buildLambdaFunction = (
       Multiplicity.ONE,
     );
     lambdaFunction.expressionSequence[0] = getAllVersionsFunction;
+  } else if (
```

Comment:
This looks good! But i feel like maybe combine the first two conditions could be a good option (person preference)
 if (options?.useAllVersionsForMilestoning ||  queryBuilderState.getAllFunction ===
    QUERY_BUILDER_SUPPORTED_GET_ALL_FUNCTIONS.GET_ALL_VERSIONS) {
      if (milestoningStereotype) {
      const getAllVersionsFunction = buildGetAllVersionsFunction(
        _class,
        Multiplicity.ONE,
      );
      lambdaFunction.expressionSequence[0] = getAllVersionsFunction;
    } else {
      throw new UnsupportedOperationError(
        `Unable to build query lamdba: getAllVersions() expects source class to be milestoned`,
      );
    }
  }

Reply:
I think i would just use switch case for all the `else if` conditions. Would be more cleaner